src/components/TranslationValueRow/index.js

```diff
@@ -0,0 +1,73 @@
+// @flow
+import React, { Component } from 'react'
+import { type Locale } from '@types/Locale'
+import { type TranslationValue } from '@types/TranslationValue'
+import { Wrapper, ContentWrapper, LocaleWrapper, Textarea } from './styles'
+
+type Props = {
+  locale: Locale,
+  translationKeyValue: TranslationValue | null,
+  onBlur: string => any
+}
+
+type State = {
+  isFocused: boolean,
+  value: string
+}
+
+class TranslationValueRow extends Component<Props, State> {
+  state = {
+    isFocused: false,
+    value: ''
+  }
+
+  onChange = (value: string) => {
+    this.setState(state => ({
+      ...state,
+      value
+    }))
+  }
+
+  toggleFocus = () => {
+    this.setState(state => ({ isFocused: !state.isFocused }))
+  }
+
+  onInputBlur = () => {
+    this.props.onBlur(this.state.value)
+    this.toggleFocus()
+  }
+
+  componentDidMount = () => {
+    const { translationKeyValue } = this.props
+
+    if (translationKeyValue) {
+      this.setState(state => ({
+        ...state,
+        value: translationKeyValue.value
+      }))
+    }
+  }
+
+  render() {
+    const { isFocused, value } = this.state
+    const { locale } = this.props
+
+    return (
+      <Wrapper>
+        <ContentWrapper isFocused={isFocused}>
+          <LocaleWrapper>{locale.name}</LocaleWrapper>
+          <Textarea
+            rows="1"
+            placeholder={`Translation for ${locale.name}`}
+            value={value}
+            onChange={e => this.onChange(e.target.value)}
```

Comment:
You should probably just do `onChange={this.onChange}` and get `event.target.value` to avoid generating a new method on every re-render.